pkg/apis/options/upstreams.go

```diff
@@ -8,7 +8,15 @@ const (
 )
 
 // Upstreams is a collection of definitions for upstream servers.
-type Upstreams []Upstream
+type Upstreams struct {
+	// ProxyRawPath will pass the raw url path to upstream allowing for url's
+	// like: "/%2F/" which would otherwise be redirected to "/"
+	ProxyRawPath bool `json:"proxyRawPath,omitempty"`
+
+	// Upstream represents the configuration for an upstream server.
+	// Requests will be proxied to this upstream if the path matches the request path.
+	Configs []Upstream `json:"configs,omitempty"`
```

Comment:
The comment should start with the name of the field, so in this case `Configs` instead of upstreams.

One quick question, what's a better name here in terms of UX, the three possibilities I can think of are `configs`, `backends` or `servers`, did you have any thoughts about why `configs`?

Reply:
How about we call the surrounding struct UpstreamConfig and this one Upstreams?
Easy enough to change.